sysdeps/atlas/generic/generic.cpp

```diff
@@ -0,0 +1,710 @@
+#include "mlibc/lock.hpp"
+#include <abi-bits/vt.h>
+#include <bits/ensure.h>
+#include <dirent.h>
+#include <errno.h>
+#include <fs.h>
+#include <limits.h>
+#include <linux/fb.h>
+#include <mlibc/all-sysdeps.hpp>
+#include <mlibc/debug.hpp>
+#include <mlibc/thread-entry.hpp>
+#include <string.h>
+#include <sys/types.h>
+#include <syscall.h>
+#include <unistd.h>
+
+namespace mlibc {
+
+void sys_libc_log(const char *message) {
+
+  int ret;
+  int errno;
+  SYSCALL_NA1(SYS_LOG_LIBC, message);
+
+  return;
+}
+
+void sys_libc_panic() {
+  mlibc::infoLogger() << "\e[31mmlibc: panic!" << frg::endlog;
+  for (;;)
+    ;
+}
+
+int sys_tcb_set(void *pointer) {
+
+  int ret;
+  int errno;
+  SYSCALL_NA1(SYS_TCB_SET, pointer);
+
+  mlibc::infoLogger() << "Got tcb_set return " << ret << frg::endlog;
+  return ret;
+}
+
+int sys_anon_allocate(size_t size, void **pointer) {
+
+  int errno = sys_vm_map(NULL, size, PROT_EXEC | PROT_READ | PROT_WRITE,
+                         MAP_ANONYMOUS, -1, 0, pointer);
+
+  return errno;
+}
+
+int sys_anon_free(void *pointer, size_t size) {
+  // TODO
+  return 0;
+}
+
+#ifndef MLIBC_BUILDING_RTDL
+void sys_exit(int status) {
+
+  int ret;
+  int errno;
+  SYSCALL_NA1(SYS_EXIT, status);
+}
```

Comment:
The compiler will complain about this `noreturn` function returning, so it might be worthwhile to add `__builtin_unreachable()` here.